docker: podman-friendly image locations

tiborsimko opened the pull request:
Adds fully qualified canonical locations of container images, making the container technology setup podman-friendly.

Closes reanahub/reana#729.

tiborsimko committed:
docker: podman-friendly image locations

Adds fully qualified canonical locations of container images, making the
container technology setup podman-friendly.

Closes reanahub/reana#729.

mdonadoni reviewed (commented):
We have some troubles with two links:
1. `https://gitter.im/reanahub/reana` is now a redirect to `https://app.gitter.im/#/room/#reanahub_reana:gitter.im`, so we should probably switch to the second one?
2. `https://twitter.com/reanahub/lists/reana-developers` now returns 404 (but only in the CI, probably due to the User Agent used by `awesome_bot`), should we white list this one?

tiborsimko commented:
>     1. `https://gitter.im/reanahub/reana` is now a redirect to `https://app.gitter.im/#/room/#reanahub_reana:gitter.im`, so we should probably switch to the second one?

I guess we can start deprecating Gitter and switching to the new Mattermost team, but this will take some time...
 
>     2. `https://twitter.com/reanahub/lists/reana-developers` now returns 404 (but only in the CI, probably due to the User Agent used by `awesome_bot`), should we white list this one?

IIRC Twitter now requires to create an account in order to read tweets, so this URL won't be verifiable.

I guess we should make "two exceptions" for the above two URLs. I'll modify the scripts to white-list them.

tiborsimko committed:
ci: white-list Gitter/Twitter URLs

White-lists Gitter and Twitter URLs for the link checker to make the
tests pass due to anonymous link checker not being able to access
Twitter content anymore. (And Gitter redirections to Matrix.)
